src/Roslyn.Suppressions/Roslyn.Suppressions/Settings.Cache/SettingsCache.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Roslyn.Suppressions.SettingsFile;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions.Settings.Cache
+{
+    internal class SettingsCache : ISettingsCache
+    {
+        private readonly ISuppressedIssuesFileStorage fileStorage;
+        private readonly ConcurrentDictionary<string, IEnumerable<SonarQubeIssue>> settingsCollection;
+
+
+        public SettingsCache(ILogger logger) : this(new SuppressedIssuesFileStorage(logger), new SettingsSingleton())
+        {
+
+        }
+
+        internal SettingsCache(ISuppressedIssuesFileStorage fileStorage, ISettingsSingleton singleton)
```

Comment:
I don't understand what the purpose of the `ISettingsSingleton` interface is. We aren't going to be creating multiple instances of `SettingsCache`.

If you need to get hold of the internal dictionary for testing purposes, you could change the `internal` constructor so the tests can directly pass in the dictionary to use:

`public SettingsCache(ILogger logger);
internal SettingsCache(ILogger logger, ConcurrentDictionary<...> settingsMap)`

Or add an internal accessor that exposes the dictionary to the tests e.g. 
`internal /* for testing */ IReadOnlyDictionary<...> SettingsMap { get { returns _settingsMap; } }`

Reply:
As we discussed I have removed the Singleton.